src/parser/symtable.ts

```diff
@@ -1625,7 +1622,7 @@ export class SymbolTable {
                 this.visitExpr(asyncFor.target);
                 this.visitExpr(asyncFor.iter);
                 this.SEQ(this.visitStmt, asyncFor.body);
-                if (asyncFor.orelse) this.SEQ(this.visitStmt, asyncFor.orelse);
+                if (asyncFor.orelse.length !== 0) this.SEQ(this.visitStmt, asyncFor.orelse);
```

Comment:
```suggestion
                if (asyncFor.orelse.length !== 0) {
                    this.SEQ(this.visitStmt, asyncFor.orelse);
                }
```

Reply:
I don't know I thought I did this everywhere. there are more cases.

Reply:
Yeah agreed.